docs/DEVICE-CONNECTIVITY.md

```diff
@@ -1,142 +1,483 @@
+I/O bus controllers (e.g., PCI controllers) be designated for assignment to applications.
+That part of the configuration can also change in run-time.
+
+Any network port configuration changes which might affect the connectivity to the controller
+requires care to avoid permanently losing connectivity to the controller and become unmanageable
+as a result. Thus, it is required to perform testing as part of a configuration change
+and have a way to fall back to a working port configuration. This is accomplished by a logic
+to test connectivity to the controller, and persistently maintaining a list of network
+configurations, including the latest and possibly one or more fallback configurations.
+
+## Network adapters
+
+Network adapter denotes network configuration (MTU, MAC, IP, VLAN, etc.) and logical attributes
+(logical & interface names, cost, usage) attached to a physical network port.
+As show in the diagram below, there is a hierarchy of different types of network adapters
+and higher-level objects attached to them, starting with the physical ports on one side and
+ending with the EVE management plane and applications on the other side:
+
+![port-hierarchy](./images/eve-network-ports.png)
+
+Network adapters use logical labels for identification and for reference within the hierarchical
+layers. Unlike higher-level objects like apps or network instances that use UUIDs, these labels
+are user-defined, simpler human-readable strings, which are only required to be unique within
+the edge-node, not globally.
+
+### Physical network ports
+
+EVE supports ethernet NICs, WiFi modules and cellular modems.
+Support for [CAN devices is a work in-progress](https://zededa.atlassian.net/wiki/spaces/EVE/pages/2163539969/Introducing+CAN+support+on+EVE).
+
+Physical network IO devices are listed as part of the device model. This should include
+all devices that will be used by EVE (for management) or by applications (directly or shared).
+IO device is referenced either using a physical address that it was given on the bus through
+which it is connected (PCI or USB), or via the name assigned by the Linux kernel to the
+network interface representing the device on the data-plane. The latter approach is potentially
+not stable and therefore not recommended. Interface name can change after a reboot if multiple
+devices of the same type are initialized in a different order. This is quite common with cellular
+modems or with ethernet NICs operated by different drivers loaded at random order.
+
+The prerequisite for using a given network IO device inside the EVE domain (i.e. not directly
+assigned to an application), is that the Linux kernel used by EVE is already shipped with
+a suitable (open-source) driver. It may be necessary to adjust the [kernel](https://github.com/lf-edge/eve-kernel)
+options to enable building of the driver.
+For cellular modems this is not required, however. Cellular modems are fairly complex
```

Comment:
Aren't there cases when EVE (and/or eve-kernel) need to be updated to support additional modems?

Reply:
I added clarification that we support only modems connected over USB (and not over PCI). In that case, the modem interface is unified and we control any modem with the same set of drivers.

Reply:
Ok